q2_gg2/_methods.py

```diff
@@ -67,6 +75,180 @@ def _classify(tree, features):
+                           metadata, strict=False):
+    """Split a distance matrix into its paired components
+
+    Parameters
+    ----------
+    distance_matrix : skbio.DistanceMatrix
+        The distance matrix to split into paired components
+    metadata : pd.DataFrame
+        The metadata to use to inform the split. We are assuming the DataFrame
+        contains a column called 'paired_sample', which describes pairing
+        relationships, and a column called 'preparation' which describes
+        whether the sample is '16S' or 'WGS'.
+    strict : bool
+        If true, limit the paired components to only samples which exist in
+        both preparations, e.g., a sample's 'paired_sample' also exists in the
+        distance matrix
+
+    Notes
+    -----
+    The sample metadata may be a superset of the samples in the distance matrix
+
+    Raises
+    ------
+    KeyError
+        If 'paired_sample' or 'preparation' do not exist as columns in the
+        DataFrame
+    ValueError
+        If more than 2 or less than 2 unique 'preparation's, with samples in
+        the distance matrix, exists
+
+    Returns
+    -------
+    skbio.DistanceMatrix, skbio.DistanceMatrix
+        A tuple where the first distance matrix are the 16S data, and the
+        second distance matrix are the WGS data.
+    """
+
+
+# RAISE ERRORS
+    if 'paired_sample' not in metadata.columns:
+        raise KeyError('Metadata must include paired_sample')
+    elif 'preparation' not in metadata.columns:
+        raise KeyError('Metadata must include preparation')
+    elif any(metadata['preparation'].isnull()):
+        raise ValueError('Missing preparations')
+    elif len(set(metadata['preparation'])) != 2:
+        raise ValueError('Must have exactly 2 unique prepartions')
+    elif any(i not in list(metadata.index) for i in distance_matrix.ids):
+        raise ValueError('Sample data missing IDs in distance matrix')
+
+# Separate 16s and WGS
+    _16s = metadata[metadata['preparation'] == '16S']
+    _sg = metadata[metadata['preparation'] == 'WGS']
+
+# filter ID's that are in distance matrix
+    _16s = _16s[_16s.index.isin(distance_matrix.ids)]
+    _sg = _sg[_sg.index.isin(distance_matrix.ids)]
+
+# check strict
+    if strict is True:
+        distance_matrix_16s = distance_matrix.filter(
+                              _16s[_16s.index.isin(
+                                _sg['paired_sample'])].index)
+        distance_matrix_sg = distance_matrix.filter(
+                             _sg[_sg.index.isin(
+                                _16s['paired_sample'])].index)
+    else:
+        distance_matrix_16s = distance_matrix.filter(_16s.index)
+        distance_matrix_sg = distance_matrix.filter(_sg.index)
+
+# return tuple
+    return (distance_matrix_16s, distance_matrix_sg)
+
+
+def _compute_effect_size(distance_matrix_16s, distance_matrix_wgs,
+                         metadata, columns=None, _max_level_by_category=5):
```

Comment:
Can `_max_level_by_category` be renamed `max_level_by_category`? Prefixing by `_` typically denotes a private variable in Python, but this isn't private